CMakeLists.txt

```diff
@@ -268,22 +268,27 @@ else (APPLE)
          set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} /arch:SSE")
          set(CMAKE_C_FLAGS    "${CMAKE_C_FLAGS} /arch:SSE")
       endif (USE_SSE)
-      set(CMAKE_CXX_FLAGS_DEBUG          "/MT /permissive- /std:c++14 /W4 /Zi /Ob0 /Od /RTC1 /wd4996")
-      set(CMAKE_CXX_FLAGS_RELEASE        "/MT /permissive- /std:c++14 /W4 /O2 /Ob2 /DNDEBUG /DQT_NO_DEBUG /DQT_NO_DEBUG_OUTPUT")
-      set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "/MT /permissive- /std:c++14 /W4 /Zi /O2 /Ob1 /DNDEBUG /DQT_NO_DEBUG /DQT_NO_DEBUG_OUTPUT /wd4996")
+      set(CMAKE_CXX_FLAGS_DEBUG          "/MT /permissive- /std:c++20 /W4 /Zi /Ob0 /Od /RTC1 /wd4996")
+      set(CMAKE_CXX_FLAGS_RELEASE        "/MT /permissive- /std:c++20 /W4 /O2 /Ob2 /DNDEBUG /DQT_NO_DEBUG /DQT_NO_DEBUG_OUTPUT")
+      set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "/MT /permissive- /std:c++20 /W4 /Zi /O2 /Ob1 /DNDEBUG /DQT_NO_DEBUG /DQT_NO_DEBUG_OUTPUT /wd4996")
       set(CMAKE_C_FLAGS_DEBUG            "/MT /W4 /Zi /Ob0 /Od /RTC1")
       set(CMAKE_C_FLAGS_RELEASE          "/MT /W4 /O2 /Ob2 /DNDEBUG")
       set(CMAKE_C_FLAGS_RELWITHDEBINFO   "/MT /W4 /Zi /O2 /Ob1 /DNDEBUG")
       set(CMAKE_EXE_LINKER_FLAGS         "/DYNAMICBASE:NO /NODEFAULTLIB:LIBCMT")
    else (MSVC)
-      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11 -D_UNICODE -DUNICODE")
       if (MINGW)
+         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++2a -D_UNICODE -DUNICODE")
          # -mno-ms-bitfields see #22048
          set(CMAKE_CXX_FLAGS         "${CMAKE_CXX_FLAGS} -mno-ms-bitfields")
          if (NOT BUILD_64)
              set(CMAKE_EXE_LINKER_FLAGS "-Wl,--large-address-aware")
          endif (NOT BUILD_64)
       else (MINGW)
+         if (ARCH_TYPE=_x64)
+            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++20 -D_UNICODE -DUNICODE")
+         else (ARCH_TYPE=_x64) # arm*
```

Comment:
This line looks a little peculiar: is there meant to be an `else if` or like a different arch_type? diff arch type or an else if? Could be fine, just double-checking

Reply:
It seems a bit wacky, but apparently does do the trick